lib/admin_block.rb

```diff
@@ -0,0 +1,200 @@
+
+require_relative 'room_wave1'
+require_relative 'reservation_wave1'
+require_relative 'block'
+require 'date'
+
+
+module Hotel
+  NUMBER_OF_ROOMS = 20
+
+  class Admin
+    attr_reader :list_of_rooms,:all_reservations, :block_reservation
+
+    def initialize
+
+      @all_reservations = []
+      @block_reservation = []
+      @list_of_rooms = create_rooms
+    end
+
+    # As an administrator, I can access the list of all of the rooms in the hotel
+    def get_list
+      rooms = []
+      @list_of_rooms.each do |room|
+        rooms << room.number
+      end
+      return rooms
+
+    end
+
+    #- As an administrator, I can reserve a room for a given date range
+    def make_reservation(start_date,end_date)
+      room_number = @list_of_rooms.sample
+      reservation = Hotel::Reservation.new(start_date,end_date,room_number)
+      @all_reservations << reservation
+      return @all_reservations
+    end
+
+    # - As an administrator, I can access the list of reservations for a specific date
+    def get_reservations(date)
+
+      if date.class!= Date
+        raise ArgumentError.new("invalid date")
+      end
+
+      @all_reservations.each do |res|
+        list =[]
+        if res.start_date <= date && res.end_date > date
+          list << res
+        end
+        return list
+      end
+
+    end
+
+    # As an administrator, I can view a list of rooms that are not reserved for a given date range
+    def list_of_available_rooms(date)
+      if date.class!= Date
+        raise ArgumentError.new("invalid date")
+      end
+
+      reserved_list = get_reservations(date)
+      reserved_rooms = []
+      reserved_list.each do |reservation|
+        reserved_rooms << reservation.room_number
+      end
+      return @list_of_rooms - reserved_rooms
+    end
+
+    # To reserve available rooms for a given date range also eliminates rooms that are in reserved in a block
+    def reserve_available_rooms(start_date,end_date)
+
+      rooms_available_range = get_available_rooms(start_date,end_date)
+
+      if rooms_available_range.empty?
+        raise Exception.new("No rooms are available")
+      else
+        list_room_block = get_block_room(start_date,end_date)
+        rooms_available_range = rooms_available_range - list_room_block
+        availability = rooms_available_range[0]
+
+        reservation = Hotel::Reservation.new(start_date,end_date,availability)
+        @all_reservations << reservation
+        return @all_reservations
+      end
+    end
+
+    # - As an administrator, I can get the total cost for a given reservation
```

Comment:
This is a cost for a date range, not a given reservation.  Small distinct difference.